csharp/src/Microsoft.ML.OnnxRuntime/ProviderOptions.cs

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.ML.OnnxRuntime
+{
+    /// <summary>
+    /// Holds the options for configuring a CUDA Execution Provider instance
+    /// </summary>
+    public class OrtCUDAProviderOptions : SafeHandle
+    {
+        internal IntPtr Handle
+        {
+            get
+            {
+                return handle;
+            }
+        }
+
+        #region Constructor
+
+        /// <summary>
+        /// Constructs an empty OrtCUDAProviderOptions instance
+        /// </summary>
+        public OrtCUDAProviderOptions() : base(IntPtr.Zero, true)
+        {
+            NativeApiStatus.VerifySuccess(NativeMethods.OrtCreateCUDAProviderOptions(out handle));
+        }
+
+        #endregion
+
+        #region Public Methods
+
+        /// <summary>
+        /// Updates  the configuration knobs of OrtCUDAProviderOptions that will eventually be used to configure a CUDA EP
+        /// Please refer to the following on different key/value pairs to configure a CUDA EP and their meaning:
+        /// https://github.com/microsoft/onnxruntime/blob/gh-pages/docs/reference/execution-providers/CUDA-ExecutionProvider.md
```

Comment:
>https://github.com/microsoft/onnxruntime/blob/gh-pages/docs/reference/execution-providers/CUDA-ExecutionProvider.md [](start = 12, length = 115)

FYI, i believe this doc page will be published at [https://www.onnxruntime.ai/docs/reference/execution-providers/CUDA-ExecutionProvider.html](https://www.onnxruntime.ai/docs/reference/execution-providers/CUDA-ExecutionProvider.html). @natke would that be a more stable link to reference?

Reply:
That seems right. I use the link where it will be published.